schema/home/home_page_artist_module.js

```diff
@@ -19,25 +19,21 @@ import {
 export const HomePageArtistModuleTypes = {
   SUGGESTED: {
     description: 'Artists recommended for the specific user.',
-    fetch: (accessToken, userID) => {
+    fetch: function fetch(accessToken, userID) {
       return gravity.with(accessToken)(`user/${userID}/suggested/similar/artists`);
     },
-    display: (accessToken, userID) => {
-      const fetch = HomePageArtistModuleTypes.SUGGESTED.fetch;
-      return Promise.resolve(!!(accessToken && userID)).then(display => {
-        return display && fetch(accessToken, userID).then(results => {
-          return results.length > 0;
-        });
-      });
+    display: function display(accessToken, userID) {
+      if (!accessToken || !userID) {
+        return Promise.resolve(false);
+      }
+      // Performing a full `fetch` instead of a count, so the next `fetch` call will be cached.
+      return this.fetch(accessToken, userID).then(results => results.length > 0);
     },
-    resolve: (accessToken, userID) => {
+    resolve: function resolve(accessToken, userID) {
```

Comment:
I luv it, but this was the only way I figured out to have a function that would have `this` bound to the object it’s in and have `this.fetch` work. But as I removed `fetch` in the end, I was also able to get rid of this oldskool mess.